### PR TITLE
Update links to point to dev docs

### PR DESCRIPTION
**Pull request by joethei**
- replace submission instruction links with links to dev docs.
- removes the `plugin-review.md` file
- update the link in the plugin template to point to plugin guidelines in the dev docs
- add link to theme guidelines in theme template
- replace open source & advertisement policy with link to dev policies.

### CODE REVIEW

**Comment by ericaxu** [comment]
I feel this is pretty safe. Do you need approval from both of them?

**Comment by joethei** [comment]
No, it would be nice if Licat could take a quick look at the last one though.

**Comment by ericaxu** [comment]
> * replace open source & advertisement policy with link to dev policies.

Last one being "replace open source & advertisement policy with link to dev policies", right? If so I'll let Licat know this is blocked on him.

**Comment by joethei** [comment]
corect

File: README.md
Comment (lishid) [inline README.md:17]: Can we make this its own heading so it's emphasized?

**Review by lishid — APPROVED** [review]

**Pull request merged**
